sdk/nodejs/x/automation/stack.ts

```diff
@@ -104,6 +111,39 @@ export class Stack {
                 throw new Error(`unexpected Stack creation mode: ${mode}`);
         }
     }
+    // Try for up to 10s to start tailing the file, invoking the callback once per line. Returns
+    // a promise for a callback to invoke to stop tailing the file.
+    private async readLines(path: string, callback: (event: EngineEvent) => void): Promise<TailFile> {
+        let n = 0;
+        let fileExists = fs.existsSync(path);
+        while (!fileExists) {
+            if (n > 100) {
+                throw new Error("Log file never found.");
+            }
+            await delay(100);
+            fileExists = fs.existsSync(path);
+            n++;
+        }
+        const eventLogTail = new TailFile(path, { startPos: 0 });
+        eventLogTail
+            .on("tail_error", (err) => {
+                console.error("TailFile had an error!", err);
+                throw err;
+            })
+            .start()
```

Comment:
It may not be a big win - but looking at the docs for `start()`, it appears it (a) returns a promise and (b) rejects that promise only if the file doesn't exist yet.  That would allow simplifying this function a fair bit - and using the call to `start` as the thing that is retried in a loop instead of `fs.existsSync`.  It would also allow avoiding the need for the `catch` here, and just `await`ing the `start` call so that any error from that naturally bubbles up (after the timeout if you do the first suggestion above) into the `Promise<TailFile>`  return value.